agent/comms.go

```diff
@@ -57,6 +57,7 @@ func (a *orbAgent) unsubscribeGroupChannels() {
 		if token := a.client.Unsubscribe(rpcFromCoreTopic); token.Wait() && token.Error() != nil {
 			a.logger.Warn("failed to unsubscribe to group channel", zap.String("topic", channel), zap.Error(token.Error()))
 		}
+		a.logger.Info("completed RPC unsubscription to group", zap.String("topic", rpcFromCoreTopic))
```

Comment:
what will really help us as we start working with users is to log the group name instead of the channelID. I see that we do send the name as part of the group RPC, so let's start using that more instead of just ID.

Reply:
We're looping through groupChannels, which is a string[] right now of channel IDs, but instead it could be an array that included the name of the group, not just the channel. We maintain this array when group subscription RPCs arrive.

Reply:
Cool, i think i'm on the right path

Reply:
I'll land some change in a few minutes

Reply:
`{
  "policy_state": {
    "c361089c-28ba-43dc-9b70-4758f1898f92": {
      "state": "running",
      "datasets": [
        "d03f3222-f56b-4dcd-b094-d8c8bf91e1b9"
      ]
    },
    "e02b2f07-d4b8-44b7-9269-23e7e9c995be": {
      "state": "running",
      "datasets": [
        "52b55949-cfba-48d5-8952-17968b614ae4"
      ]
    }
  },
  "backend_state": {
    "pktvisor": {
      "state": "running"
    }
  },
  "subscribed_groups": {
    "agent-edit": {
      "group_channel": "3dc2da1d-f4f9-4ff3-96a4-9139097331c0"
    },
    "agent-test-3": {
      "group_channel": "4bc0a1c5-6981-4098-ae98-f6550cee3973"
    }
  }
}`

Reply:
looks good except why don't we make it consistent with policy_state and make "subscribe_groups" keyed by group_id, then have "name" and "channel" keys inside that sub object. and while we are here, it would be great to add "name" key to the policy state too, that would be very useful for debugging as well.

Reply:
```
{
  "policy_state": {
    "c361089c-28ba-43dc-9b70-4758f1898f92": {
      "name": "policy",
      "state": "running",
      "datasets": [
        "d03f3222-f56b-4dcd-b094-d8c8bf91e1b9"
      ]
    },
    "e02b2f07-d4b8-44b7-9269-23e7e9c995be": {
      "name": "policy2",
      "state": "running",
      "datasets": [
        "52b55949-cfba-48d5-8952-17968b614ae4"
      ]
    }
  },
  "backend_state": {
    "pktvisor": {
      "state": "running"
    }
  },
  "subscribe_groups": {
    "4167f8b3-7746-410f-a86b-2a6bf3a201e4": {
      "group_name": "agent-test-3",
      "group_channel": "4bc0a1c5-6981-4098-ae98-f6550cee3973"
    },
    "80abf802-0815-439f-99e2-98848fc249ad": {
      "group_name": "agent-edit",
      "group_channel": "3dc2da1d-f4f9-4ff3-96a4-9139097331c0"
    }
  }
}
```

Reply:
yes! except "subscribed_groups" and just "name" and "channel" without the group_

Reply:
should be "subscribe_groups" or "subscribed_groups"??

Reply:
subscribed